resources/assets/lib/profile-page/historical.tsx

```diff
@@ -59,28 +79,40 @@ function updateTicks(chart: LineChart<Date>, data?: ChartData[]) {
   }
 }
 
-interface Props extends ExtraPageProps {
-  beatmapPlaycounts: BeatmapPlaycountJson[];
-  currentMode: GameMode;
-  pagination: ProfilePagePaginationData;
-  scoresRecent: ScoreJson[];
-}
-
-export default class Historical extends React.PureComponent<Props> {
+@observer
+export default class Historical extends React.Component<ExtraPageProps> {
   private readonly chartRefs = {
     monthly_playcounts: React.createRef<HTMLDivElement>(),
     replays_watched_counts: React.createRef<HTMLDivElement>(),
   };
   private readonly charts: Partial<Record<ChartSection, LineChart<Date>>> = {};
   private readonly id = `users-show-historical-${nextVal()}`;
 
+  @computed
+  private get monthlyPlaycountsData() {
+    return convertUserDataForChart(this.props.controller.state.user.monthly_playcounts);
+  }
+
+  @computed
+  private get replaysWatchedCountsData() {
+    return convertUserDataForChart(this.props.controller.state.user.replays_watched_counts);
+  }
+
+  constructor(props: ExtraPageProps) {
+    super(props);
+
+    makeObservable(this);
+  }
+
   componentDidMount() {
     $(window).on(`resize.${this.id}`, this.resizeCharts);
-    this.updateCharts();
-  }
+    disposeOnUnmount(this, autorun(this.updateCharts));
 
-  componentDidUpdate() {
-    this.updateCharts();
+    disposeOnUnmount(this, autorun(() => {
+      Object.values(this.charts).forEach((chart) => {
+        updateTicks(chart);
```

Comment:
`updateTicks` needs to run once with the right data before the chart is actually drawn (or chart redrawn after `updateTicks`).